tests/test_encounter/test_window_handling.py

```diff
@@ -1,177 +1,267 @@
+        self.ground_location = GroundLocation(0, 0, 0, u.deg, u.km)
+        self.vtw_attitude = Attitude(
+            np.random.rand(5),
+            np.random.rand(5),
+            np.random.rand(5),
+            np.random.rand(5),
+            u.deg,
+            self.ground_location
+        )
+        self.ow_attitude = Attitude(
+            np.random.rand(1),
+            np.random.rand(1),
+            np.random.rand(1),
+            np.random.rand(1),
+            u.deg,
+            self.ground_location
+        )
+        self.test_vtw = VisibleTimeWindow(0, 1, self.vtw_attitude)
+        self.test_obw = ObservationWindow(
+            Quantity(0, u.jd2000),
+            Quantity(1, u.s),
+            Quantity(2, u.jd2000),
+            self.ground_location,
+            self.ow_attitude
+        )
+
+    def test_initialization(self):
+        self.assertIsInstance(self.window_handler, WindowHandler)
+        self.assertListEqual(self.window_handler._window_data, [])
+
+    def test_add_window_raises_error_for_non_window(self):
+        self.assertRaises(TypeError, self.window_handler.add_window,
+                          "not a window")
+
+    def test_add_window_alters_window_data_with_vtw(self):
+        self.window_handler.add_window(self.test_vtw)
+        self.assertListEqual(self.window_handler._window_data, [self.test_vtw])
+
+    def test_add_window_alters_window_data_with_ow(self):
+        self.window_handler.add_window(self.test_obw)
+        self.assertListEqual(self.window_handler._window_data, [self.test_obw])
+
+    def test_add_window_enforces_same_window_type(self):
+        self.window_handler.add_window(self.test_vtw)
+        self.assertRaises(TypeError, self.window_handler.add_window,
+                          self.test_obw)
+
+    def test_str(self):
+        self.window_handler.add_window(self.test_vtw)
+        self.window_handler.add_window(self.test_vtw)
+        self.assertEqual(str([self.test_vtw, self.test_vtw]),
+                         str(self.window_handler))
+
+    def test_repr(self):
+        self.window_handler.add_window(self.test_vtw)
+        self.window_handler.add_window(self.test_vtw)
+        self.assertEqual(f"WindowHandler({repr(self.test_vtw)}, {repr(self.test_vtw)})",
+                         repr(self.window_handler))
+
+    def test_len_returns_correct_length(self):
+        self.assertEqual(len(self.window_handler), 0)
+        self.window_handler.add_window(self.test_vtw)
+        self.assertEqual(len(self.window_handler), 1)
+        self.window_handler.add_window(self.test_vtw)
+        self.assertEqual(len(self.window_handler), 2)
+
+    def test_iteration(self):
+        self.window_handler.add_window(self.test_vtw)
+        self.window_handler.add_window(self.test_vtw)
+        self.assertListEqual([window for window in self.window_handler],
+                             self.window_handler._window_data)
+
+    def test_indexing_with_integers(self):
```

Comment:
Look at refactoring to remove the duplicate code between this method and the next.

Reply:
May be clunky to resolve.